pkg/rehearse/jobs.go

```diff
@@ -123,21 +125,82 @@ func ExecuteJobs(toBeRehearsed map[string][]prowconfig.Presubmit, prNumber int,
 		}
 	}
 
-	if len(rehearsals) > 0 {
-		if err := rehearsalConfigs.Create(); err != nil {
-			return fmt.Errorf("failed to prepare rehearsal ci-operator config ConfigMap: %v", err)
+	if len(rehearsals) == 0 {
+		logger.Warn("No job rehearsals")
+		return nil
+	}
+	if err := rehearsalConfigs.Create(); err != nil {
+		return fmt.Errorf("failed to prepare rehearsal ci-operator config ConfigMap: %v", err)
+	}
+	pjs := make(map[string]struct{}, len(rehearsals))
+	for _, job := range rehearsals {
+		created, err := submitRehearsal(job, refs, logger, pjclient)
+		if err != nil {
+			logger.WithError(err).Warn("Failed to execute a rehearsal presubmit")
+			continue
 		}
-		for _, job := range rehearsals {
-			created, err := submitRehearsal(job, refs, logger, pjclient)
-			if err != nil {
-				logger.WithError(err).Warn("Failed to execute a rehearsal presubmit")
-			} else {
-				logger.WithFields(pjutil.ProwJobFields(created)).Info("Submitted rehearsal prowjob")
-			}
+		logger.WithFields(pjutil.ProwJobFields(created)).Info("Submitted rehearsal prowjob")
+		pjs[created.Name] = struct{}{}
+	}
+	selector := fmt.Sprintf("%s=%d", rehearseLabel, prNumber)
```

Comment:
I'd rather see this done by building the selector and using it's toString method

Reply:
Use Selector from [apimachinery/pkg/labels](https://godoc.org/k8s.io/apimachinery/pkg/labels#Selector) maybe?